js_modules/dagit/packages/ui/src/components/Tooltip.tsx

```diff
@@ -36,6 +36,23 @@ export const Tooltip: React.FC<Props> = (props) => {
 
   const [isOpen, setIsOpen] = React.useState<undefined | boolean>(undefined);
 
+  const divRef = React.useRef<any>(null);
+
+  React.useLayoutEffect(() => {
+    let listener: null | ((e: MouseEvent) => void) = null;
+    if (isOpen && useDisabledButtonTooltipFix) {
+      listener = (e: MouseEvent) => {
+        if (!divRef.current?.contains(e.target as HTMLDivElement)) {
+          setIsOpen(false);
+        }
```

Comment:
Look upward from `e.target` instead of downward from `divRef`? Then you're just looking up the ancestry list instead of through the entire tree below `divRef`? Maybe nbd since tooltip targets are generally small.

Or maybe `compareDocumentPosition`?

Reply:
Yeah I did consider that but the tree isn't that big so it seems like early optimization. I'm using `contains` though which theoretically probably works that way under the hood?